unit_testing/batch_processing/test_batch_processing.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+# pytest unit tests to validate the reults of the batch_processing.sh script
+
+import os
+import pathlib
+import csv
+
+import pytest
+
+from spinalcordtoolbox.utils.sys import sct_dir_local_path
+
+
+@pytest.mark.skipif(
+    not os.getenv('BATCH_PROCESSING_CI_JOB'),  # This environment variable should be set by the CI workflow file
+    reason="Run only for batch processing CI job"
+)
+@pytest.mark.parametrize("csv_filepath,row,pos,rel_tolerance",
+                         [("t2/csa_c2c3.csv",    -1, 5, 1e-14),
+                          ("t2s/csa_gm.csv",     -1, 5, 1e-14),
+                          ("t2s/csa_wm.csv",     -1, 5, 1e-14),
+                          ("mt/mtr_in_wm.csv",   -1, 7, 1e-8),
+                          ("dmri/fa_in_cst.csv", -1, 7, 1e-8),
+                          ("dmri/fa_in_cst.csv", -2, 7, 1e-7)])
+def test_batch_processing_results(csv_filepath, row, pos, rel_tolerance):
+    """Ensure that new batch_processing.sh results are within a certain tolerance of the cached baseline results."""
+    sct_dir = pathlib.Path(sct_dir_local_path())
+    csv_filepath_old = sct_dir / "unit_testing/batch_processing/cached_results" / csv_filepath
+    csv_filepath_new = sct_dir / "sct_example_data" / csv_filepath
+    assert csv_filepath_old.is_file(), f"{csv_filepath_old} not present. Please check the SCT installation."
+    assert csv_filepath_new.is_file(), f"{csv_filepath_new} not present. Was batch_processing.sh run beforehand?"
+
+    with open(csv_filepath_old, newline='') as csvfile:
+        reader = csv.reader(csvfile, delimiter=',')
+        metric_value_old = float([row for row in reader][row][pos])  # Row/position varies depending on metric
+
+    with open(csv_filepath_new, newline='') as csvfile:
+        reader = csv.reader(csvfile, delimiter=',')
+        metric_value_new = float([row for row in reader][row][pos])  # Row/position varies depending on metric
+
+    assert metric_value_new == pytest.approx(metric_value_old, rel=rel_tolerance)
```

Comment:
Use `np.genfromtxt` here. I discovered it while helping Julien with https://github.com/neuropoly/spinalcordtoolbox/pull/3050#discussion_r557700055:

https://github.com/neuropoly/spinalcordtoolbox/blob/6b66ae770ec85650a74f3b201152831f72d491f0/unit_testing/test_register.py#L202-L206

Reply:
Is this only spot-checking a single entry right now?

Reply:
Nifty! Thanks for the tip. 😄

Reply:
> Is this only spot-checking a single entry right now?

These 6 values:

https://github.com/neuropoly/spinalcordtoolbox/blob/64315ccdc50eec434a391739f46a0fe177419b4b/unit_testing/batch_processing/test_batch_processing.py#L19-L24

Are mimicking these 6 values:

https://github.com/neuropoly/spinalcordtoolbox/blob/255140ade8632524122e09ea53908bb56cf10ca3/batch_processing.sh#L257-L262

Reply:
Do you think we should be checking more than 1 value per csv? Or am I misunderstanding?

At the very least, I've added a note about this in fee0953.